### PR TITLE
[docs] create a new contributors' guide

### PR DESCRIPTION
**Pull request by polarweasel**
@netlify /latest/contribute/docs/

Notes:

* syntax diagram section is ripped straight out of the readme, and needs cleanup
* added a link to the Contribute dropdown

### CODE REVIEW

File: docs/content/latest/contribute/_index.md [hunk 1]
```diff
@@ -35,23 +35,23 @@ This is the C++ code and the unit tests that comprise the core of YugabyteDB. Yo
 
 ### Docs
 
-[YugabyteDB documentation](/) uses the Hugo framework. There are two types of docs issues - infrastructure enhancements and adding content. You can [follow the steps outlined here](https://github.com/yugabyte/yugabyte-db/tree/master/docs) to run a local version of the docs site. Next, look at the [contributing guide](https://github.com/yugabyte/yugabyte-db/blob/master/docs/CONTRIBUTING.md) to make your changes and contribute them.
+[YugabyteDB documentation](/) uses the Hugo framework. There are two types of docs issues - infrastructure enhancements and adding content. You can [follow the steps outlined here](docs/) to get set up and make a contribution.
 
 ## Find an issue
 
-Issues are tagged with other useful labels as described below.
+Issues are tagged with labels as described below.
```
Comment (lizayugabyte) [inline docs/content/latest/contribute/_index.md:42]: Issues are tagged with labels, as per the following table.

File: docs/content/latest/contribute/docs/_index.md [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+
+<div class="row">
+  <div class="col-12 col-md-6 col-lg-12 col-xl-6">
+    <a class="section-link icon-offset" href="docs-checklist/">
+      <div class="head">
+        <img class="icon" src="/images/section_icons/deploy/checklist.png" aria-hidden="true" />
+        <div class="title">Checklist</div>
+      </div>
+      <div class="body">
+        Explore the whole contribution process, from clone to pull request
```
Comment (lizayugabyte) [inline docs/content/latest/contribute/docs/_index.md:25]: Explore the contribution process, from clone to pull request

File: docs/content/latest/contribute/docs/_index.md [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+  </div>
+
+  <div class="col-12 col-md-6 col-lg-12 col-xl-6">
+    <a class="section-link icon-offset" href="docs-edit/">
+      <div class="head">
+        <img class="icon" src="/images/section_icons/index/troubleshoot.png" aria-hidden="true" />
+        <div class="title">Edit the docs</div>
+      </div>
+      <div class="body">
+        Fork the repository, and edit the docs
```
Comment (lizayugabyte) [inline docs/content/latest/contribute/docs/_index.md:49]: Fork the repository and edit the docs

File: docs/content/latest/contribute/docs/docs-build.md [hunk 4]
```diff
@@ -0,0 +1,117 @@
...
+menu:
+  latest:
+    identifier: docs-build
+    parent: docs
+    weight: 2913
+isTocNested: true
+showAsideToc: true
+---
+
+## Install prerequisites
```
Comment (lizayugabyte) [inline docs/content/latest/contribute/docs/docs-build.md:17]: Prerequisites

**Review by lizayugabyte — COMMENTED** [review]
Started making comments, but then stopped.
I assume the contributors will treat this doc as a style guide of sorts. I suggest that we follow conventions (e.g. when to use different fonts, how to introduce lists, etc.)
Feel free to reassign me the ticket.

File: docs/content/latest/contribute/docs/docs-page-structure.md [hunk 5]
```diff
@@ -50,25 +66,21 @@ menu:
 | `showAsideToc`  | `false` | Should the TOC on the right be enabled |
```
Comment (ddhodge) [inline docs/content/latest/contribute/docs/docs-page-structure.md:66]: | `isTocNested`  | `false` | Should sub-sections be displayed in the TOC on the right. Set to true. (Content pages only) |

File: docs/content/latest/contribute/docs/docs-style.md [hunk 6]
```diff
@@ -0,0 +1,77 @@
...
+
+**Use second person.** Address the reader directly: "Before you start, make sure your VM is configured as follows."
+
+**Active voice.** There are rare times when a passive construction is a lot less convoluted.
+
+**Contractions.** They're good. Don't use them in a legal contract, but do use them most other places.
+
+### Headings
+
+**Sentence case.** Headings should be sentence case. For example, "Provision your cluster", not "Provision Your Cluster". Obvious exceptions include proper nouns, product names, and more.
```
Comment (ddhodge) [inline docs/content/latest/contribute/docs/docs-style.md:35]:
```suggestion
**Sentence case.** For example, "Provision your cluster", not "Provision Your Cluster". Obvious exceptions include proper nouns, product names, and more.
```

File: docs/content/latest/contribute/_index.md [hunk 7]
```diff
@@ -35,23 +35,23 @@ This is the C++ code and the unit tests that comprise the core of YugabyteDB. Yo
...
+| [kind/new-feature](https://github.com/yugabyte/yugabyte-db/issues?q=is%3Aopen+is%3Aissue+label%3Akind%2Fnew-feature) | An new feature does not exist. New features can be complex additions to the existing system, or standalone pieces. |
+| [kind/question](https://github.com/yugabyte/yugabyte-db/issues?q=is%3Aopen+is%3Aissue+label%3Akind%2Fquestion) | A question that needs to be answered. |
 
-* If you are just starting out contributing to YugabyteDB, first off welcome and thanks! Look for issues labelled [good first issue](https://github.com/yugabyte/yugabyte-db/issues?q=is%3Aopen+is%3Aissue+label%3A%22good+first+issue%22).
+* If you're just starting out contributing to YugabyteDB, first off welcome and thanks! Look for issues labelled [good first issue](https://github.com/yugabyte/yugabyte-db/issues?q=is%3Aopen+is%3Aissue+label%3A%22good+first+issue%22).
 
 * If you have already contributed and are familiar, then look for issues with the [help wanted](https://github.com/yugabyte/yugabyte-db/issues?q=is%3Aopen+is%3Aissue+label%3A%22help+wanted%22) label.
```
Comment (ddhodge) [inline docs/content/latest/contribute/_index.md:56]:
```suggestion
* If you've already contributed and are familiar, then look for issues with the [help wanted](https://github.com/yugabyte/yugabyte-db/issues?q=is%3Aopen+is%3Aissue+label%3A%22help+wanted%22) label.
```

File: docs/content/latest/contribute/docs/syntax-diagrams.md [hunk 8]
```diff
@@ -0,0 +1,417 @@
...
+menu:
+  latest:
+    identifier: docs-syntax-diagrams
+    parent: docs-edit
+    weight: 2915
+isTocNested: true
+showAsideToc: true
+---
+
+In this kind of work, the nature of your documentation change means that you'll need to modify existing [_syntax diagrams_](#syntax-diagram) or add new ones.
```
Comment (ddhodge) [inline docs/content/latest/contribute/docs/syntax-diagrams.md:18]:
```suggestion
The API documentation uses [_syntax diagrams_](#syntax-diagram). The following describes diagram maintenance and creation for the YSQL API documentation. (The YCQL documentation uses an old method for syntax diagrams; take advice from colleagues if you need to work on YCQL diagrams.)
```

File: docs/content/latest/contribute/docs/syntax-diagrams.md [hunk 9]
```diff
@@ -0,0 +1,417 @@
...
+
+> **Note:** The following covers diagram maintenance and creation for the YSQL documentation. The YCQL documentation still uses an old method for this. You must take advice from colleagues if you need to work on YCQL diagrams.
```
Comment (ddhodge) [inline docs/content/latest/contribute/docs/syntax-diagrams.md:20]:
```suggestion
```

File: docs/content/latest/contribute/docs/syntax-diagrams.md [hunk 10]
```diff
@@ -0,0 +1,417 @@
...
+
+Once you understand this area, modifying existing [_syntax diagrams_](#syntax-diagram) or adding new ones will seem to be very straightforward. However, you must embrace a fairly large mental model. This, in turn, depends on several terms of art. All this is covered in the dedicated section [Creating and maintaining syntax diagrams for YSQL Docs](#3-mental-model-and-glossary-creating-and-maintaining-syntax-diagrams-for-ysql-docs).
+
+Here is a terse step-by-step summary:
+
+### Get set up
```
Comment (ddhodge) [inline docs/content/latest/contribute/docs/syntax-diagrams.md:26]:
```suggestion
## Get set up
```

File: docs/content/latest/contribute/docs/syntax-diagrams.md [hunk 11]
```diff
@@ -0,0 +1,417 @@
...
+
+1. Download the latest RRDiagram JAR file (`rrdiagram.jar`).  See the section the [_diagram generator_](#diagram-generator).
+
+1. [_Run the generator_](#run-the-generator). This will regenerate every diagram file that is jointly specified by the [_diagram definition file_](#diagram-definition-file) and the set of all [_free-standing generated grammar-diagram pairs_](#free-standing-generated-grammar-diagram-pair). Of course, not all of these newly generated `md` files will differ from their previous versions; `git status` will show you only the ones that actually changed.
+
+### Modify or add content pages or syntax rules
```
Comment (ddhodge) [inline docs/content/latest/contribute/docs/syntax-diagrams.md:32]:
```suggestion
## Modify or add content pages or syntax rules
```

File: docs/content/latest/contribute/docs/syntax-diagrams.md [hunk 12]
```diff
@@ -0,0 +1,417 @@
...
+1. If necessary, [_run the generator_](#run-the-generator).
+1. If necessary, add [_diagram inclusion HTML_](#diagram-inclusion-HTML) to one or more of the content files that you modified.
+1. If necessary, add URL references in other content files to link to your new work. For example, you might need to update an `_index.md` file that implements a table of contents to a set of files to which you've just added one.
+1. If necessary, add (or update) the `aliases` page property in any  [_humanly typed documentation source_](#humanly-typed-documentation-source) file(s) that you relocated.
+1. Manually check the new docs page(s) and index page(s) to make sure that there are no broken links.
+
+> **Note:** There is no tool for link checking within your local git. Broken links are detected when a Yugabyte staff member periodically runs a link checker on the published doc on the Internet. The scope of this check includes _all_ of the Yugabyte properties (for example, the [Blogs site](https://blog.yugabyte.com)). Broken links that this check finds are reported and fixed manually.
+
+## 3. Mental model and glossary: creating and maintaining syntax diagrams for YSQL Docs
```
Comment (ddhodge) [inline docs/content/latest/contribute/docs/syntax-diagrams.md:87]:
```suggestion
## Mental model and glossary: creating and maintaining syntax diagrams for YSQL Docs
```

File: docs/content/latest/contribute/docs/syntax-diagrams.md [hunk 13]
```diff
@@ -0,0 +1,417 @@
...
+    weight: 2915
+isTocNested: true
+showAsideToc: true
+---
+
+In this kind of work, the nature of your documentation change means that you'll need to modify existing [_syntax diagrams_](#syntax-diagram) or add new ones.
+
+> **Note:** The following covers diagram maintenance and creation for the YSQL documentation. The YCQL documentation still uses an old method for this. You must take advice from colleagues if you need to work on YCQL diagrams.
+
+Once you understand this area, modifying existing [_syntax diagrams_](#syntax-diagram) or adding new ones will seem to be very straightforward. However, you must embrace a fairly large mental model. This, in turn, depends on several terms of art. All this is covered in the dedicated section [Creating and maintaining syntax diagrams for YSQL Docs](#3-mental-model-and-glossary-creating-and-maintaining-syntax-diagrams-for-ysql-docs).
```
Comment (ddhodge) [inline docs/content/latest/contribute/docs/syntax-diagrams.md:22]:
```suggestion
Once you understand this area, modifying existing [_syntax diagrams_](#syntax-diagram) or adding new ones will seem to be very straightforward. However, you must embrace a fairly large mental model. This, in turn, depends on several terms of art. All this is covered in the dedicated section [Creating and maintaining syntax diagrams for YSQL Docs](#mental-model-and-glossary-creating-and-maintaining-syntax-diagrams-for-ysql-docs).
```

File: docs/content/preview/integrations/hasura/graphql.md [hunk 14]
```diff
@@ -3,6 +3,7 @@ title: Benchmarking
 linkTitle: Benchmarking
 description: Using the benchmark tool with Hasura
 aliases:
+  - /latest/develop/graphql
```
Comment (polarweasel) [inline docs/content/preview/integrations/hasura/graphql.md:6]:
```suggestion
aliases:
  - /preview/develop/graphql
```

**Pull request merged**